### PR TITLE
Fix ERR_PACKAGE_PATH_NOT_EXPORTED

### LINKED ISSUE #27: ERR_PACKAGE_PATH_NOT_EXPORTED
Updating to `3.1.2` from `3.1.1` resulted in following error when using the plugin with `jiti`:
```
> jiti ./scripts/generate.ts

...\solid-primitives\node_modules\.pnpm\jiti@1.19.1\node_modules\jiti\lib\index.js:2
  throw err; /* ↓ Check stack trace ↓ */
  ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in ...\solid-primitives\site\node_modules\remark-emoji\package.json
    at new NodeError (node:internal/errors:399:5)
    at exportsNotFound (node:internal/modules/esm/resolve:361:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:641:13)
    at Module._findPath (node:internal/modules/cjs/loader:634:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1061:27)
    at Function.resolve (node:internal/modules/cjs/helpers:116:19)
    at _resolve (...\solid-primitives\node_modules\.pnpm\jiti@1.19.1\node_modules\jiti\dist\jiti.js:1:250100)
    at jiti (...\solid-primitives\node_modules\.pnpm\jiti@1.19.1\node_modules\jiti\dist\jiti.js:1:252409)
    at ...\solid-primitives\site\scripts\generate.ts:12:43 {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}

Node.js v18.15.0
 ELIFECYCLE  Command failed with exit code 1.
```

### PR DESCRIPTION
**Pull request by thetarnav**
fixes #27

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
       "import": {
         "types": "./types/index.d.ts",
         "default": "./index.js"
-      }
+      },
+      "default": "./index.js"
```
Comment (rhysd) [inline package.json:15]: Since `default` is a general fallback, `require` is better.

**Review by rhysd — CHANGES_REQUESTED** [review]

**Comment by rhysd** [comment]
Some questions:

- [`index.js`](https://github.com/rhysd/remark-emoji/blob/master/index.js) is an ESM module file. Generally just `require()`ing ESM module file causes an import error. Why does setting it as entry point of CJS work?
- How did you confirm your change is correct? I'd like to follow the same steps.


I just tried the same change in my local repository directory. The `ERR_PACKAGE_PATH_NOT_EXPORTED` was resolved by it, but I got the following error. This error is consistent with the behavior I mentioned in the first question.
  ```
  Uncaught:
  Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/rhysd/Develop/github.com/rhysd/remark-emoji/index.js not supported.
  Instead change the require of index.js in null to a dynamic import() which is available in all CommonJS modules.
      at __node_internal_captureLargerStackTrace (node:internal/errors:490:5)
      at new NodeError (node:internal/errors:399:5)
      at Module._extensions..js (node:internal/modules/cjs/loader:1282:19)
      at Module.load (node:internal/modules/cjs/loader:1113:32)
      at Module._load (node:internal/modules/cjs/loader:960:12)
      at Module.require (node:internal/modules/cjs/loader:1137:19)
      at require (node:internal/modules/helpers:121:18) {
    code: 'ERR_REQUIRE_ESM'
  }
  ```

**Comment by thetarnav** [comment]
> * [`index.js`](https://github.com/rhysd/remark-emoji/blob/master/index.js?rgh-link-date=2023-08-08T13%3A04%3A04Z) is an ESM module file. Generally just `require()`ing ESM module file causes an import error. Why does setting it as entry point of CJS work?

I'm not using CJS. I think. Unless I'm misunderstanding something. The `remark` package works fine, this one does too. Only the exports were wrong for some reason.

> * How did you confirm your change is correct? I'd like to follow the same steps.

I just directly modified the package.json in node_modules until it worked for me [here](https://github.com/solidjs-community/solid-primitives/tree/main/site). But then I just reverted to 3.1.1, instead of patching the module.

For a better test I've made this: https://github.com/thetarnav/remark-emoji-export-issue
It installs `remark-emoji@3.1.2` with this patch applied, and everything works great for both node 18, 20; and typescript 5 and 4.

I'm not entirely sure how there exports are supposed to be setup, I usually just change them until everything works. But one thing I've noticed is that adding more conditions doesn't hurt, but may help :D

**Comment by rhysd** [comment]
> I'm not using CJS. I think.

Then it is very weird because you only modified configuration for CJS in this PR.

It looks you're using TypeScript. Can you provide your tsconfig which is used for compiling scripts which imports remark-emoji?

**Comment by rhysd** [comment]
From your example (thanks for the repository for clarification), your tsconfig is

```json
{
  "compilerOptions": {
    "module": "ESNext",
    "target": "ESNext",
    "newLine": "LF",
    "moduleResolution": "node",
    "strict": true,
    "types": ["node"]
  }
}
```

https://github.com/thetarnav/remark-emoji-export-issue/blob/main/examples/tsx/tsconfig.json

The `"moduleResolution": "node"` is very legacy configuration and it only assumes CommonJS. So you're using CommonJS module resolution unexpectedly.

From https://www.typescriptlang.org/tsconfig#moduleResolution

> 'node10' (previously called 'node') for Node.js versions older than v10, which only support CommonJS require. You probably won’t need to use node10 in modern code.

You should use `node16`, `nodenext` or `bundler` which support both ESM and CJS.

**Comment by thetarnav** [comment]
Alright, switching to NodeNext works for me too. shall I close the issue and PR then?

**Comment by rhysd** [comment]
Thanks for confirming it. I'm closing this and the issue.

**Pull request closed without merging**